### ISSUE TITLE
BubbleUPnP and DMS

### ISSUE THREAD

**Issue by adler99** [anacrolix/dms#105]
Hi,

I've problems to browse large libraries with BubbleUPnP (Android Phone) and a DMS server. Talking about ~26k of mp3s. 
- DMS seems to start more `ffprobe` threads on each library reload, leading to high server loads
- I never managed to reload the library or server search the library with BubbleUPnP

How many files can DMS manage? What are you experiences? Is there any tuning possible?

Thanks!

**Comment by anacrolix** [comment]
This might be related to https://github.com/anacrolix/dms/issues/47. I suspect you need to ensure caching occurs across process restarts, and might be struggling to trigger metadata for the first use of your (very large) directory?

**Comment by adler99** [comment]
yes, I guess you need a persistent cache for large directories. I've the feeling that even when one scan completes, the next request coming in is triggering another full scan. As if there would be no caching at all...
Could you share how many files you found to work well? 
Thanks!

**Comment by anacrolix** [comment]
I've tested with a few 1000, but a few small fixes will make this scale to arbitrary levels.

**Comment by anacrolix** [comment]
Please try master for @fholzer's improvements. It may help with the problem described here.
